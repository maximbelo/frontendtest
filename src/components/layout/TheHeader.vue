<template>
  <header>
    <div class="header-content container">
      <div class="logo">
        <a href="https://www.chess.com/"
          ><img src="/images/chesscom-logo.png" alt="Chess.com logo"
        /></a>
      </div>
      <h1>ChessClickathon.</h1>
    </div>
  </header>
</template>

<script setup lang="ts"></script>

<style lang="scss" scoped>
header {
  margin: 32px 0;
}

.header-content {
  display: flex;
  align-items: end;
  justify-content: space-between;

  @media (max-width: 768px) {
    flex-direction: column;
    align-items: start;
    width: 88%;
  }

  a {
    display: flex;
  }

  img {
    height: 59px;
    width: 192px;

    @media (max-width: 768px) {
      height: 37px;
      width: 131px;
      margin-bottom: 10px;
    }
  }
}
</style>
