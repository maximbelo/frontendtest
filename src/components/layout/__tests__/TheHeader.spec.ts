import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";
import TheHeader from "@/components/layout/TheHeader.vue";

describe("TheHeader.vue tests", () => {
  test("matches the snapshot", () => {
    const wrapper = mount(TheHeader, { shallow: true });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
