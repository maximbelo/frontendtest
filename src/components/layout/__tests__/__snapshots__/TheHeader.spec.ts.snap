// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheHeader.vue tests > matches the snapshot 1`] = `
"<header data-v-c4a0481f=\\"\\">
  <div data-v-c4a0481f=\\"\\" class=\\"header-content container\\">
    <div data-v-c4a0481f=\\"\\" class=\\"logo\\"><a data-v-c4a0481f=\\"\\" href=\\"https://www.chess.com/\\"><img data-v-c4a0481f=\\"\\" src=\\"/images/chesscom-logo.png\\" alt=\\"Chess.com logo\\"></a></div>
    <h1 data-v-c4a0481f=\\"\\">ChessClickathon.</h1>
  </div>
</header>"
`;
